### PR TITLE
Add more exceptions to handle for key access denied; Handle hidden co…

### PR DESCRIPTION
**Pull request by shangxinli**
Add more exceptions to handle for key access denied; 
Handle hidden columns when checking encodings;

### CODE REVIEW

**Comment by shangxinli** [comment]
This pull request includes two changes that block plaintext columns to be read in Hive when there is 1 or more encrypted columns, even user only select plaintext columns to read. 

To reproduce this issue, here are the steps. 
1. Create two columns c1(plaintext), c2(encrypted) in tbl1. 
2. Run 'select c1 from tbl1'

Then the exception will be reported. 

The AccessControlException is returned from Hadoop KMS, and AccessDeniedException is returned from AWS KMS. Hence both are required.

File: parquet-hadoop/src/main/java/org/apache/parquet/crypto/InternalFileDecryptor.java [hunk 1]
```diff
@@ -230,8 +232,8 @@ public InternalColumnDecryptionSetup setColumnCryptoMetadata(ColumnPath path, bo
           // No explicit column key given via API. Retrieve via key metadata.
           try {
             columnKeyBytes = keyRetriever.getKey(keyMetadata);
-          } 
-          catch (KeyAccessDeniedException e) {
+          }
+          catch (KeyAccessDeniedException | AccessControlException | AccessDeniedException e) {
```
Comment (ggershinsky) [inline parquet-hadoop/src/main/java/org/apache/parquet/crypto/InternalFileDecryptor.java]: keyRetriever is your code. You can catch AccessControlException | AccessDeniedException e there, and throw new KeyAccessDeniedException(e)

File: parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordReader.java [hunk 2]
```diff
@@ -188,7 +189,11 @@ private void checkDeltaByteArrayProblem(FileMetaData meta, Configuration conf, B
       // this is okay if not using DELTA_BYTE_ARRAY with the bug
       Set<Encoding> encodings = new HashSet<Encoding>();
       for (ColumnChunkMetaData column : block.getColumns()) {
-        encodings.addAll(column.getEncodings());
+        try {
+          encodings.addAll(column.getEncodings());
+        } catch (HiddenColumnException e) {
+          LOG.warn("Bypass column {} because it is hidden column so encodings cannot be read", column);
+        }
```
Comment (ggershinsky) [inline parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordReader.java]: Sounds good. Please remove the InternalFileDecryptor.java changes, and I'll merge this pr.

File: parquet-hadoop/src/main/java/org/apache/parquet/crypto/InternalFileDecryptor.java (hunk 1 above)
Comment (ggershinsky) [reply parquet-hadoop/src/main/java/org/apache/parquet/crypto/InternalFileDecryptor.java]: by your code, I mean its always a custom implementation of the DecryptionKeyRetriever interface. - which can catch AccessControlException | AccessDeniedException e there, and throw new KeyAccessDeniedException(e).

**Pull request closed without merging**